File: solo/puzzles/easy/HorseRacingDuals.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main()
{
    int n;
    std::cin >> n; std::cin.ignore();
    std::vector<int> v;
    for (int i = 0; i < n; i++) {
        int pi;
        std::cin >> pi; std::cin.ignore();
        v.push_back(pi);
    }
    std::sort(v.begin(), v.end());
    
    int min = 10000000;
    for(int i = 1; i < v.size(); ++i)
    {
        int diff = std::abs(v[i] - v[i - 1]);
        if (diff < min)
        {
            min = diff;
        }
    }
    std::cout << min << std::endl;
}

File: solo/puzzles/easy/horse-racing-duals/horse_racing_duals.rs
use std::io;

macro_rules! parse_input {
   ($x:expr, $t:ident) => ($x.trim().parse::<$t>().unwrap())
}

fn main() {
   let mut input_line = String::new();
   io::stdin().read_line(&mut input_line).unwrap();
   let n = parse_input!(input_line, i32);
   let mut v: Vec<i32> = Vec::new();
   for i in 0..n as usize {
      let mut input_line = String::new();
      io::stdin().read_line(&mut input_line).unwrap();
      let pi = parse_input!(input_line, i32);
      v.push(pi);
   }

   v.sort();

   let mut min: i32 = i32::MAX;
   for i in 1..n as usize {
      let diff = i32::abs(v[i] - v[i - 1]);
      if diff < min {
         min = diff;
      }
   }

   println!("{}", min);
}
